org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/commands/CheckFilePatternCommand.java

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.contentmodel.commands;
+
+import java.nio.file.FileSystems;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+
+import org.eclipse.lemminx.services.IXMLDocumentProvider;
+import org.eclipse.lemminx.services.extensions.commands.ArgumentsUtils;
+import org.eclipse.lemminx.services.extensions.commands.IXMLCommandService.IDelegateCommandHandler;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+public class CheckFilePatternCommand implements IDelegateCommandHandler {
+
+	public static final String COMMAND_ID = "xml.check.file.pattern";
+	private IXMLDocumentProvider documentProvider;
+
+	public CheckFilePatternCommand(IXMLDocumentProvider documentProvider) {
+		this.documentProvider = documentProvider;
+	}
+
+	@Override
+	public Object executeCommand(ExecuteCommandParams params, SharedSettings sharedSettings,
+			CancelChecker cancelChecker) throws Exception {
+		String pattern = ArgumentsUtils.getArgAt(params, 0, String.class);
```

Comment:
We should have an uri too as argument. The idea is to validate that a given xml file matches à given pattern.

Reply:
Use https://github.com/eclipse/lemminx/blob/a3058b72a344c89c854e44104c87faa3142b2a66/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/PathPatternMatcher.java#L42 to check that given uri matches the given pattern.